@here/harp-lrucache/lib/LRUCache.ts

```diff
@@ -159,6 +153,26 @@ export class LRUCache<Key, Value> {
         this.evict();
     }
 
+    /**
+     * Updates the size of all elements in this cache. If their aggregated size is larger than the
+     * capacity, items will be evicted until the cache shrinks to fit the capacity.
+     */
+    checkCapacity(): void {
+        let size = 0;
+        const sizeFunction = this.sizeFunction;
+
+        let entry = this.m_newest;
```

Comment:
Any reason you iterate from newest to oldest rather than just iterate over the internal `m_map`?

Reply:
It seemed to be the least overhead compared to the alternatives, of which there seem to be many. Do you have a specific loop in mind?

Reply:
The other functions use `m_map.forEach()`, you could also use `for (let [key, value] of this.m_map)`

Reply:
As I see it, none of your suggested approaches are comparable in performance or simplicity. Saving a few TypeScript chars to replace them with many JS/machine code bytes does not look like an improvement to me.